homeassistant/components/gogogate2/cover.py

```diff
@@ -55,64 +55,42 @@ def __init__(
         """Initialize the object."""
         unique_id = cover_unique_id(config_entry, door)
         super().__init__(config_entry, data_update_coordinator, door, unique_id)
-        self._api = data_update_coordinator.api
-        self._is_available = True
+        self._attr_supported_features = SUPPORT_OPEN | SUPPORT_CLOSE
+        self._attr_device_class = (
+            DEVICE_CLASS_GATE if self.door.gate else DEVICE_CLASS_GARAGE
+        )
 
     @property
     def name(self):
         """Return the name of the door."""
-        return self._get_door().name
+        return self.door.name
 
     @property
     def is_closed(self):
         """Return true if cover is closed, else False."""
-        door_status = self._get_door_status()
+        door_status = self.door_status
         if door_status == DoorStatus.OPENED:
             return False
         if door_status == DoorStatus.CLOSED:
             return True
```

Comment:
```suggestion
        if self.door_status == DoorStatus.OPENED:
            return False
        if self.door_status == DoorStatus.CLOSED:
            return True
```

Reply:
I saved the result of the property in this case because it does a few lookups under the hood and I did not want to pay for the overhead of doing it multiple times

Reply:
Oh I see, you are doing an API call in the `door_status` property. Maybe this should be refactored to have a local copy of that value, as properties should not be used when I/O is necessary, see [here](https://developers.home-assistant.io/docs/core/entity/cover#properties)

Reply:
I don't think it's a matter of doing I/O in this case, right @bdraco?

Reply:
It's not doing I/O just a few hash lookups that I'd prefer to do only once.